### PR TITLE
Added the task property window in a new PR to remove the build files …

### PR DESCRIPTION
**Pull request by virajparimi**
I am submitting a fresh PR and have removed the Ravel.pro.user file and the previous commits as well.

### CODE REVIEW

**Comment by kisaacs** [comment]
Per Slack conversation, please update this so that task selection and timeline panning still behave as they did previously. Thanks!

File: src/mainwindow.h [hunk 1]
```diff
@@ -30,12 +30,15 @@
 #include <QStack>
 #include <QList>
 #include <QVector>
+#include "traditionalvis.h"
```
Comment (kisaacs) [inline src/mainwindow.h]: Where is this used?

File: src/mainwindow.h [hunk 2]
```diff
@@ -30,12 +30,15 @@
...
 
 class Event;
 class Trace;
 class VisWidget;
 class VisOptions;
+class TraditionalVis;
```
Comment (kisaacs) [inline src/mainwindow.h]: Where is this used?

File: src/timelinevis.cpp [hunk 3]
```diff
@@ -140,7 +140,11 @@ void TimelineVis::mousePressEvent(QMouseEvent * event)
     mousePressed = true;
     rightPressed = false;
     if (event->button() == Qt::RightButton)
+    {
         rightPressed = true;
+        if( Qt::ControlModifier && event->modifiers() )
```
Comment (kisaacs) [inline src/timelinevis.cpp]: The spacing of this line doesn't match the others.

File: src/timelinevis.h [hunk 4]
```diff
@@ -41,6 +41,7 @@ class TimelineVis : public VisWidget
     virtual void mouseDoubleClickEvent(QMouseEvent * event);
     void leaveEvent(QEvent * event);
     virtual void rightDrag(QMouseEvent * event) { Q_UNUSED(event); }
+    virtual void rightClickEvent(QMouseEvent * event){ Q_UNUSED(event); }
```
Comment (kisaacs) [inline src/timelinevis.h]: The spacing of this line doesn't match the others.

File: src/traditionalvis.cpp [hunk 5]
```diff
@@ -84,6 +83,23 @@ void TraditionalVis::setTrace(Trace * t)
...
+    for( QMap<Event *, QRect>::Iterator evt = drawnEvents.begin();
+         evt != drawnEvents.end(); evt++ )
+        if( evt.value().contains(mousex, mousey) )
+        {
+            task_property_event = evt.key();
+            break;
+        }
+    emit taskPropertyDisplay(task_property_event);
+}
+
```
Comment (kisaacs) [inline src/traditionalvis.cpp:102]: Please fix the spacing throughout so it matches the rest of the project.

File: src/mainwindow.h (hunk 1 above)
Comment (virajparimi) [reply src/mainwindow.h]: Removed

File: src/mainwindow.cpp [hunk 6]
```diff
@@ -204,6 +208,27 @@ void MainWindow::selectEntities(QList<int> entities)
...
+            if(fnc.key() == function_id)
+            {
+                QString units = RavelUtils::RavelUtils::getUnits((*trc)->units);
+                taskwindow = new TaskPropertyWindow(this, event, fnc.value(), units);
+                break;
+            }
+        }
+    }
+    taskwindow->show();
+}
```
Comment (kisaacs) [inline src/mainwindow.cpp:230]: Please fix spacing to match the rest of the project.

**Review by kisaacs — CHANGES_REQUESTED** [review]

**Review by kisaacs — APPROVED** [review]

**Pull request merged**